src/install/dracut-install.c

```diff
@@ -1574,21 +1600,40 @@ static void find_suppliers_for_sys_node(struct kmod_ctx *ctx, Hashmap *suppliers
                         closedir(d);
                 }
                 strncat(node_path, "/..", 3); // Also find suppliers of parents
+                char *parent_path = realpath(node_path, NULL);
+                if (parent_path != NULL)
+                        if (hashmap_put(suppliers, parent_path, parent_path) < 0)
+                                free(parent_path);
         }
 }
 
 static void find_suppliers(struct kmod_ctx *ctx)
 {
         _cleanup_fts_close_ FTS *fts;
-        char *paths[] = { "/sys/devices/platform", NULL };
+        char *paths[] = { "/sys/devices", NULL };
```

Comment:
@bdrung, I think it's this change causing the slowdown on AMD64. Can you test with just this line reverted?

I saw a number of devices under `/sys/devices/pci0000:00/` and `/sys/devices/LNXSYSTM:00/`, and saw a devlink `pci:0000:0d:00.0--pci:0000:0d:00.1` (`snd_hda_intel` and `amdgpu`) on my AMD64 system. So I assumed things like those might also exist on ARM64 and made the device search more general here. But I don't see anything outside `/sys/devices/platform` on device-tree systems (just `/sys/devices/soc` that resolves to crypto modules).

Reply:
I've reverted this change, which should avoid slowing down the x86 cases. But I expect it might eventually come up again if kernel people start using devlinks on ACPI systems without declaring module dependencies.